rust/connlib/tunnel/src/tests/sut.rs

```diff
@@ -674,11 +670,13 @@ impl TunnelTest {
             ClientEvent::ResourcesChanged { .. } => {
                 tracing::warn!("Unimplemented");
             }
-            ClientEvent::TunInterfaceUpdated {
-                dns_by_sentinel, ..
-            } => {
+            ClientEvent::TunInterfaceUpdated(config) => {
+                if self.client.inner().dns_by_sentinel == config.dns_by_sentinel {
+                    tracing::error!("Emitted `TunInterfaceUpdated` without changing DNS servers");
+                }
```

Comment:
@jamilbk @conectado 

With these lines here, we now assert that we don't emit `TunInterfaceUpdated` unless something actually changed. This does not yet include the routes (which was the bug in #6387) but it will be much easier to add the routes to that after this PR.